api/scripts/database/drop-database.js

```diff
@@ -12,6 +12,14 @@ url.pathname = '/postgres';
 
 const client = new PgClient(url.href);
 
+const disconnectAllSessionsButMineQuery =
+  `SELECT pg_terminate_backend(pg_stat_activity.pid)
+  FROM pg_stat_activity
+  WHERE pg_stat_activity.datname = '${DB_TO_DELETE_NAME}'
+  AND pid <> pg_backend_pid();`;
+
+(async () => { await client.query_and_log(disconnectAllSessionsButMineQuery); })();
```

Comment:
C'est quoi l'intention de ce `(async () => { await …` ?

Si c'est pour lancer la requête sans attendre la résolution,
```suggestion
client.query_and_log(disconnectAllSessionsButMineQuery);
```
marcherait aussi (a priori c'est juste équivalent à ce qui est écrit), mais on s'expose à la possibilité d'une `unhandled rejection`. Et je ne vois pas pourquoi on voudrait ne pas attendre la résolution ici.

Si c'est pour ignorer une éventuelle erreur, mieux vaut utiliser proprement `.then`/`.catch`.

Mais plus simplement, dans ce cas-ci j'ai l'impression qu'on peut fusionner les deux commandes SQL en un seul appel.

```suggestion
client.query_and_log(
 `SELECT pg_terminate_backend(pg_stat_activity.pid)
  FROM pg_stat_activity
  WHERE pg_stat_activity.datname = '${DB_TO_DELETE_NAME}'
  AND pid <> pg_backend_pid();
  DROP DATABASE ${DB_TO_DELETE_NAME};`)
  .then(() => {
	console.log('Database dropped');
	…
```

Reply:
Merci @jonathanperret 
Je vais me pencher un jour sérieusement sur les promesses plutôt que de taper des trucs juqu'à ce que linter passe au vert :smile:

Reply:
Zut, ça marche pas !
`DROP DATABASE cannot run inside a transaction block`

Reply:
OK, du coup rester sur deux appels distincts au client mais il faut les séquencer correctement. Par exemple :

```
async function dropDatabase(client) {
	await client.query_and_log(
	  `SELECT pg_terminate_backend(pg_stat_activity.pid)
	  FROM pg_stat_activity
	  WHERE pg_stat_activity.datname = '${DB_TO_DELETE_NAME}'
	  AND pid <> pg_backend_pid()`
    );

	await client.query_and_log(`DROP DATABASE ${DB_TO_DELETE_NAME}`);
}

…

dropDatabase(client)
	.then(() => {
		console.log('Database dropped');
		…
```

Reply:
@octo-topi Par ailleurs, je viens de voir dans la description de cette PR qu'en cas d'échec il y avait un `UnhandledPromiseRejectionWarning` — d'ailleurs en local je constate que l'exécution de `npm run db:delete` ne se termine même pas dans ce cas !

La cause en est ce `throw` : https://github.com/1024pix/pix/blob/6e7e5486ffb6e41928229a5b55f28da938fcd647/api/scripts/database/drop-database.js#L26-L28

En effet on intercepte bien le rejet de la promesse avec le `catch` mais ensuite on crée de nouveau une promesse rejetée avec ce `throw`, qui elle n'est pas interceptée.

Il faudrait remplacer ce `throw` par quelque chose comme :

```
      console.error('Error:', error.message);
      process.exit(1);
```

Je pense que ça irait bien dans cette PR-ci aussi non ?

Reply:
Merci pour les apprentissages !
Parfois, oublier une PR quelques jours aiude à prendre du recul.
Je pense être sur une fausse piste, on n'a pas besoin de supprimer la BDD en fait, juste de supprimer les objets.
Je ferme cette PR et en ouvrirait un autre lundi qui adresse ce besoin spécifique.